### PR TITLE
c-hyper: fix two memory leaks in `Curl_http`.

### PR DESCRIPTION
**Pull request by nnethercote**
These commits fix the memory leaks in #10803.

### CODE REVIEW

File: docs/KNOWN_BUGS [hunk 1]
```diff
@@ -119,12 +119,6 @@ problems may have been fixed or changed somewhat since this was written.
 
 1. HTTP
 
-1.1 hyper memory-leaks
```
Comment (bdaehlie) [inline docs/KNOWN_BUGS:122]: Probably need to renumber the two entries below and also delete/renumber in the index.

Comment (bagder) [reply docs/KNOWN_BUGS:122]: Nah, the script that converts that to show on the website handles gaps just fine. We prefer gaps than having to reorder every time we add/remove.

=> https://curl.se/docs/knownbugs.html

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Comment by nnethercote** [comment]
@bagder: In every other GitHub project I've contributed to, accepted PRs end up in the "merged" state, and the "closed" state is used for rejected PRs. But you're doing it in a different way which leaves accepted PRs marked as "Closed with unmerged commits". There is also a message above "[bagder](https://github.com/bagder) closed this in https://github.com/curl/curl/commit/c61dd5fed23525726270044f3baae2e767d87bcf", which refers to a single commit but there were three commits in this PR.

Is there a reason for this? As a newbie to the project it surprised me and confused me a bit. Perhaps you don't want merges in the commit history? If that's the case you can allow [rebase merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-rebasing-for-pull-requests). Or is there another reason?

**Comment by jay** [comment]
[That's right.](https://github.com/curl/curl/wiki/push-access-guidelines#no-merge-commits-on-master) Most PRs with pieced changes are condensed into a single commit. The commit message is always amended to meet our guidelines then rebased on upstream/master (upstream: this repo). The author is preserved and the committer name is the person pushing the commit (another guideline). The committer name is _not_ GitHub.

**Comment by bagder** [comment]
The commit that gets mentioned in the PR is the one that uses the `closes` keyword.

We do this to maintain commit message style, squashed commits and linear history. The commits are done with the command line and GitHub offers no better way to signal this in the PR. I keep pointing this out to them and hope that one day we get a `merged` keyword.

**Pull request closed without merging**